test/ShallowWrapper-spec.jsx

```diff
@@ -2684,6 +2684,18 @@ describe('shallow', () => {
   });
 
   describe('lifecycleExperimental', () => {
+    it('throws when used with disableLifecycleMethods in invalid combinations', () => {
+      expect(() => shallow(<div />, {
+        lifecycleExperimental: true,
+        disableLifecycleMethods: true,
+      })).to.throw(/same value/);
+
+      expect(() => shallow(<div />, {
+        lifecycleExperimental: false,
+        disableLifecycleMethods: false,
+      })).to.throw(/same value/);
```

Comment:
let's also add tests where both are omitted, and where either aren't a boolean (ie, it should throw if a non-boolean non-undefined value is provided)